test/unit/account/states_test.rb

```diff
@@ -153,11 +165,11 @@ def test_suspend_master
 
     account_deleted_recently = accounts[0]
     account_deleted_recently.schedule_for_deletion!
-    account_deleted_recently.update_attribute(:deleted_at, (Account::States::PERIOD_BEFORE_DELETION.ago + 1.day))
+    account_deleted_recently.update_attribute(:state_changed_at, (Account::States::PERIOD_BEFORE_DELETION.ago + 1.day))
```

Comment:
Should we also assert the new state, since `state_changed_at` can mean more things now?

Reply:
What you are asking me to add is that the method `schedule_for_deletion!`actually makes the state `scheduled_for_deletion`, which is a good test to add, sure, but it is unrelated to the fact that `state_changed_at` means more things now 🤔 
What you are asking me to test is this:
https://github.com/3scale/porta/blob/27e408560463ebc4245538037ef08b5ed82f73c5/app/models/account/states.rb#L89-L91

And what I am testing here is this:
https://github.com/3scale/porta/blob/27e408560463ebc4245538037ef08b5ed82f73c5/app/models/account/states.rb#L42-L46

Reply:
I understand that. I just wanted to assert that it was indeed `schedule_for_deletion!` that caused `state_changed_at` new value. You can also do `account.expects(:update_attributes).with(...)`

Reply:
Wait, I thought this comment was in https://github.com/3scale/porta/pull/276#discussion_r237437379
This is completely different, this is to test the response of `deleted_since` 😄

Reply:
This code is testing this:
https://github.com/3scale/porta/blob/088a699ea7135ab2264ba04d256139b0f959cb44/test/unit/account/states_test.rb#L163-L180
I think it is easier to understand if you see the whole test 😄 

What it means to test there has nothing to do with `state_changed_at` being updated correctly 😄
it just means to test the method `deleted_since`, only. Which just means returning the accounts in the DB that were deleted before X time. That's why I am setting that time in here in the test in the DB for those accounts.

Reply:
However, this other test: https://github.com/3scale/porta/pull/276/files#r237437379
is the one that means to test that `state_changed_at` is updated correctly when the state is changed.
I will comment there about it 😄

Reply:
@Martouta, please ignore my comment. I was again completely tricked by GH. My distraction.